Update for recent ESPHome releases

nagyrobi opened the pull request:
- change switch `restore_state` to `restore_mode` as the former configuration option is not supported anymore
- change onboard_led to be a `status_led` light platform, so it can be double purposed as both ESPHome Status LED and heating indicator
- add some entity icons, device class
- API password is not supported (deprecated) it needs encryption only

nagyrobi committed:
Update for recent ESPHome releases

- change switch restore_state to restore_mode as the former configuration option is not supported anymore
- change `onboard_led` to be a status_led light platform, so it can be double purposed as both ESPHome Status LED and heating indicator
- add some entity icons, device class

nagyrobi committed:
Update api config options

- API password is not supported anymore, it needs encryption

nagyrobi commented:
@jensweimann

jensweimann commented on esphome/esphome_smarttherm.yaml:
```diff
@@ -175,6 +179,7 @@ binary_sensor:
         pullup: True
     name: ${unit_name}_active
     id: power_active
+    device_class: power
```
I don't think that this device_class is right here. Its meant for current measuring in watts: https://developers.home-assistant.io/docs/core/entity/sensor/#:~:text=SensorDeviceClass.POWER
I suggest is to remove this line

nagyrobi replied:
Nope, see: https://www.home-assistant.io/integrations/binary_sensor/#device-class

Binary sensors have device class, which sets the icon and the name (language specific) for the state corresponding to true/false (like open/closed, locked/unlocked powered on/powered off).

In this case, device class power: on means power detected, off means no power, with appropriate state translated to each language, plus a plug icon with state coloring.

jensweimann replied:
Ok, thank you for contributing.

jensweimann reviewed (approved):
Looks good to me

nagyrobi commented:
Bump

jensweimann commented:
I don't have permissions to accept the PR. @Strooom merge please

nagyrobi committed:
Update esphome_smarttherm.yaml

remove api/ota/etc as these are standard ESPHome config options to be seen in the doc.
ota will soon change a bit, so no need to update this doc again, for something otherwise unrelated directly to this project

Strooom commented:
Ok, I'll take a look at it.

nagyrobi commented:
Bump
